torchgeo/datasets/utils.py

```diff
@@ -184,97 +200,134 @@ def download_radiant_mlhub_collection(
     collection.download(output_dir=download_root, api_key=api_key)
 
 
-class BoundingBox(Tuple[float, float, float, float, float, float]):
```

Comment:
Previously we subclasses from tuple. Now we define our own object from scratch. One consequence of this is that you can no longer pass a `BoundingBox` directly into an R-tree index, you have to cast it to a tuple first. We _could_ easily support this, but it requires defining the `__len__` method and having it always return 6. Personally I don't think it makes sense for a bounding box to have a "length", so I didn't do this.